### ISSUE TITLE
Econv behaviour for gb18030

### ISSUE THREAD

**Issue by LillianZ** [jruby/jcodings#39]
Should the last two lines should be equal? Thanks!

``` java
EConv econv = TranscoderDB.open("UTF-8", "gb18030", 0);

byte[] src = "Lašas".getBytes("UTF-8");
byte[] dest = new byte["Lašas".getBytes("gb18030").length];

econv.convert(src, new Ptr(0), 6, dest, new Ptr(0), dest.length, 0);
System.out.println(Arrays.toString(dest)); 
// [76, 97, -127, 48, 18, 56, 97, 115]
System.out.println(Arrays.toString("Lašas".getBytes("gb18030"))); 
// [76, 97, -127, 48, -108, 56, 97, 115]
```

**Comment by headius** [comment]
Assuming the encoding of these characters is the same in Ruby, the GB18030 bytes appear to match:

```
[] ~/projects/jruby $ ruby -e 'p "Lašas".encode("gb18030").bytes'
[76, 97, 129, 48, 148, 56, 97, 115]
```

I do not have an explanation for why the Java GB18030 encoder produces different output.

**Comment by headius** [comment]
Based on [this online converter](https://r12a.github.io/app-encodings/), we also match:

```
$ ruby -e 'p "Lašas".encode("gb18030").bytes.map{|i| i.to_s(16)}'
["4c", "61", "81", "30", "94", "38", "61", "73"]
```

![image](https://user-images.githubusercontent.com/10135/96791061-65a52300-13bd-11eb-8171-fcc6e4da67c3.png)

I would say the Java encoder is in error here.

**Comment by headius** [comment]
Actually now I see that the Java `getBytes` matches Ruby but the manually transcoded result is not correct in your example.

I made this into a test class and I believe the latest jcodings should match. Perhaps you are running against an old version?

```
$ java -cp ../jcodings/target/jcodings.jar:. Blah
[76, 97, -127, 48, -108, 56, 97, 115]
[76, 97, -127, 48, -108, 56, 97, 115]
```

```java Blah.java
import org.jcodings.*;
import org.jcodings.transcode.*;
import java.util.*;

public class Blah {
public static void main(String[] args) throws Throwable {
EConv econv = TranscoderDB.open("UTF-8", "gb18030", 0);

byte[] src = "Lašas".getBytes("UTF-8");
byte[] dest = new byte["Lašas".getBytes("gb18030").length];

econv.convert(src, new Ptr(0), 6, dest, new Ptr(0), dest.length, 0);
System.out.println(Arrays.toString(dest)); 
// [76, 97, -127, 48, 18, 56, 97, 115]
System.out.println(Arrays.toString("Lašas".getBytes("gb18030"))); 
// [76, 97, -127, 48, -108, 56, 97, 115]
}
}
```

**Comment by headius** [comment]
Possibly fixed by @k77ch7 in 408210ce852febb2959f2bcdc460f2c91c195117. In any case, it's no longer broken.

**Comment by LillianZ** [comment]
Yes, I was using an old version, thanks for helping me debug!
